libmscore/cmd.cpp

```diff
@@ -140,7 +141,8 @@ void Score::endCmd(bool rollback)
 
       if (MScore::debugMode)
             qDebug("===endCmd() %d", undo()->current()->childCount());
-      bool noUndo = (undo()->current()->childCount() <= 1);       // nothing to undo?
+      // nothing to undo if current undo macro has only the SaveState commands for each score
+      bool noUndo = (undo()->current()->childCount() <= scoreList().size());
```

Comment:
Not 100%, no, although it "should" be so.  After all, we *did* put those SaveState commands there, and it's hard to imagine what might have removed them.  And in the current source we check for a childCount() of 1, but we don't check that this is the SaveState for the current score, either.

But sure, better, I suppose, would be to check the actual commands.  But then, if there *is* something else there, that would be pretty unfortunate, as it isn't clear how that should happen or what we should do if it does - what would have happened to our SaveState commands?

Anyhow, I'm certainly open to other solutions.